Source/VMware.vSphereDSC/VMware.vSphereDSC.psm1

```diff
@@ -140,6 +140,88 @@ class VMHostBaseDSC : BaseDSC
     }
 }
 
+[DscResource()]
+class VMHostPowerPolicySettings : VMHostBaseDSC
+{
+    <#
+    .DESCRIPTION
+
+    Host setting for power policy
+    #>
+    [DscProperty()]
+    [System.Int32] $PowerPolicy = 2
+
+    $powerProfiles =  @{
+        "High Performance" = 1 ;
+        "Balanced"         = 2 ;
+        "Low Power"        = 3 ;
+        "Custom"           = 4
+    }
+
+    [void] Set()
+    {
+        $this.ConnectVIServer()
+        $vmHost = $this.GetVMHost()
+
+        $this.UpdatePowerPolicy($vmHost)
+    }
+
+    [void] UpdatePowerPolicy($vmHost)
+    {
+        $vmHostCurrentPowerPolicy = $vmHost.ExtensionData.config.PowerSystemInfo.CurrentPolicy
+        $shouldUpdateVMHostPowerPolicy = $this.ShouldUpdateVMHostPowerPolicy($vmHostCurrentPowerPolicy)
+
+        if (!$shouldUpdateVMHostPowerPolicy)
```

Comment:
Inverse if

        if (!$shouldUpdateVMHostPowerPolicy)
        {
                    $powerSystem = Get-View $vmHost.ExtensionData.ConfigManager.PowerSystem
        Update-HostPowerPolicy -PowerSystem $powerSystem -PowerPolicy $this.PowerPolicy        }